### ISSUE TITLE
Fix S1144 FP: Diagnostic doesn't respect reflection with `DynamicallyAccessedMembers` attribute

### ISSUE THREAD

**Issue by dotjpg3141** [SonarSource/sonar-dotnet#9379]
### Description

Diagnostic S1144 doesn't respect reflection even though the generic parameter is annotated with the `DynamicallyAccessedMembers` attribute.

Related: #7774

### Repro steps

This is a minified repro. In my actuall use case I called [`UseMiddleware`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.builder.usemiddlewareextensions.usemiddleware?view=aspnetcore-8.0).

```csharp
using System.Diagnostics.CodeAnalysis;

namespace SonarCubeRepro
{
    internal static class Program
    {
        static void Main()
        {
            var instance = CreateInstance<Foo>();
            Console.WriteLine( instance.Number );
        }

        public static T CreateInstance<[DynamicallyAccessedMembers( DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors )] T>()
        {
            var instance = (T)Activator.CreateInstance( typeof( T ), 42 )!;
            return instance;
        }

        private sealed class Foo
        {
            public Foo( int number )
            {
                Number = number;
            }

            public int Number { get; set; }
        }
    }
}
```

### Expected behavior

No warning.

### Actual behavior

Warning S1144

### Known workarounds

Supress the issue

### Related information

SonarAnalyzer.CSharp Version=9.26.0.92422

* C#/VB.NET Plugins version
* Visual Studio version: 17.9.6
* MSBuild / dotnet version: 8.0.204
* SonarScanner for .NET version (if used)
* Operating System: Windows 10

**Comment by zsolt-kolbay-sonarsource** [comment]
Thank you for reporting the issue. Confirmed as False Positive.

**Comment by zsolt-kolbay-sonarsource** [comment]
@dotjpg3141 It turns out fixing the issue for generic method calls [seriously degrades the analyzer performance](https://github.com/SonarSource/sonar-dotnet/pull/9406#issuecomment-2161049389), so we opted not to fix this scenario and instead document it as a False Positive.
The analyzer can still be prompted to ignore a type or a member by decorating it with the `DynamicallyAccessedMembersAttribute`: either by decorating it with the [System.Diagnostics.CodeAnalysis.DynamicallyAccessedMembersAttribute](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.dynamicallyaccessedmembersattribute) (available in .NET 5.0+) or by your own custom attribute named `DynamicallyAccessedMembersAttribute` (in .NET Framework projects).
This change will be available in the next release (9.28).
